gateway/sensor.go

```diff
@@ -68,22 +67,88 @@ func init() {
 
 // Validate ensures all parts of the config are valid.
 func (conf *Config) Validate(path string) ([]string, error) {
+	for _, d := range conf.Devices {
+		if d.DecoderPath == "" {
+			return nil, resource.NewConfigValidationError(path,
+				errors.New("decoder path is required"))
+		}
+		switch d.JoinType {
+		case "ABP":
+			return d.validateABPAttributes(path)
+		case "OTAA", "":
+			return d.validateOTAAAttributes(path)
+		default:
+			return nil, resource.NewConfigValidationError(path,
+				errors.New("join type is OTAA or ABP"))
```

Comment:
```suggestion
				errors.New("join type must be OTAA or ABP"))
```

Reply:
consider adding the note about the default, might be helpful for someone who hits this error and isn't sure where to go from there